scene/2d/path_2d.cpp

```diff
@@ -201,7 +631,46 @@ void Path2D::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("get_debug_sampling_enabled"), &Path2D::get_debug_sampling_enabled);
+
+	ClassDB::bind_method(D_METHOD("set_debug_custom_enabled", "enabled"), &Path2D::set_debug_custom_enabled);
+	ClassDB::bind_method(D_METHOD("get_debug_custom_enabled"), &Path2D::get_debug_custom_enabled);
+
+	ClassDB::bind_method(D_METHOD("set_debug_custom_color", "debug_custom_color"), &Path2D::set_debug_custom_color);
+	ClassDB::bind_method(D_METHOD("get_debug_custom_color"), &Path2D::get_debug_custom_color);
+
+	ClassDB::bind_method(D_METHOD("set_debug_custom_sampling_enabled", "enabled"), &Path2D::set_debug_custom_sampling_enabled);
+	ClassDB::bind_method(D_METHOD("get_debug_custom_sampling_enabled"), &Path2D::get_debug_custom_sampling_enabled);
+
+	ClassDB::bind_method(D_METHOD("set_debug_custom_sample_interval", "interval"), &Path2D::set_debug_custom_sample_interval);
+	ClassDB::bind_method(D_METHOD("get_debug_custom_sample_interval"), &Path2D::get_debug_custom_sample_interval);
+
+	ClassDB::bind_method(D_METHOD("set_debug_custom_fish_bones_enabled", "enabled"), &Path2D::set_debug_custom_fish_bones_enabled);
+	ClassDB::bind_method(D_METHOD("get_debug_custom_fish_bones_enabled"), &Path2D::get_debug_custom_fish_bones_enabled);
+
+	ClassDB::bind_method(D_METHOD("set_debug_custom_fish_bones_interval", "interval"), &Path2D::set_debug_custom_fish_bones_interval);
+	ClassDB::bind_method(D_METHOD("get_debug_custom_fish_bones_interval"), &Path2D::get_debug_custom_fish_bones_interval);
+
 	ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "curve", PROPERTY_HINT_RESOURCE_TYPE, "Curve2D", PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_EDITOR_INSTANTIATE_OBJECT), "set_curve", "get_curve");
+
+	ADD_GROUP("Debug", "debug_");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "debug_enabled"), "set_debug_enabled", "get_debug_enabled");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "debug_sampling_enabled"), "set_debug_sampling_enabled", "get_debug_sampling_enabled");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "debug_custom_enabled"), "set_debug_custom_enabled", "get_debug_custom_enabled");
+	ADD_PROPERTY(PropertyInfo(Variant::COLOR, "debug_custom_color"), "set_debug_custom_color", "get_debug_custom_color");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "debug_custom_sampling_enabled"), "set_debug_custom_sampling_enabled", "get_debug_custom_sampling_enabled");
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "debug_custom_sample_interval", PROPERTY_HINT_RANGE, "0.1,10,0.001,or_greater"), "set_debug_custom_sample_interval", "get_debug_custom_sample_interval");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "debug_custom_fish_bones_enabled"), "set_debug_custom_fish_bones_enabled", "get_debug_custom_fish_bones_enabled");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "debug_custom_fish_bones_interval", PROPERTY_HINT_RANGE, "1,10,1,or_greater"), "set_debug_custom_fish_bones_interval", "get_debug_custom_fish_bones_interval");
+
+	ADD_SIGNAL(MethodInfo("curve_changed"));
```

Comment:
I get why it exists, but exposing a new signal to the API of one of the most commonly used nodes in the engine... It warrants its own PR, in my opinion. Tread carefully.

Reply:
It exists on Path3D but I can just remove it.

Path2D and Path3D are half the time in a messy desync anyway so the damage is already done for their api.